packages/loader/container-loader/src/container.ts

```diff
@@ -1547,6 +1547,12 @@ export class Container
 		const supportGetSnapshotApi: boolean =
 			this.mc.config.getBoolean("Fluid.Container.UseLoadingGroupIdForSnapshotFetch") ===
 				true && this.service?.policies?.supportGetSnapshotApi === true;
+
+		// Load the pending local virtualized state
+		if (pendingLocalState?.loadingGroupSnapshots !== undefined) {
+			this.storageAdapter.loadingGroupIds = pendingLocalState?.loadingGroupSnapshots;
```

Comment:
Feeding this into the base snapshot will be tricky as we will need to properly process all the ops before doing so.